implemented #17

Alls-DeV opened the pull request:
implemented [this](https://github.com/seclerp/rider-monogame/issues/17) issue

Commit:
Add SpriteFont XML Schema Provider

Added `SpriteFontXmlSchemaProvider` class to handle XML schema for SpriteFont, which includes schema-related operations like caching and fetching the schema from the given location. The schema provider is now registered in the `plugin.xml` file. The XML Schema definition (XSD) for SpriteFont configuration file has been added to the resources directory. These changes enable better handling and validation of SpriteFont XML content in the Monogame plugin.

seclerp commented on src/rider/main/kotlin/me/seclerp/rider/plugins/monogame/spritefont/SpriteFontXmlSchemaProvider.kt:
```diff
@@ -0,0 +1,74 @@
+
+class SpriteFontXmlSchemaProvider : XmlSchemaProvider() {
+    companion object {
+        private val logger = getLogger<SpriteFontXmlSchemaProvider>()
```
It's better to use `com.intellij.openapi.diagnostic.logger<T>`:
```suggestion
        private val logger = logger<SpriteFontXmlSchemaProvider>()
```

`getLogger<T>()` just calls `logger<T>()` under the hood. It exists for some Rd codebase historical reasons.

seclerp commented on src/rider/main/kotlin/me/seclerp/rider/plugins/monogame/spritefont/SpriteFontXmlSchemaProvider.kt:
```diff
@@ -0,0 +1,74 @@
+
+    private val schemaFileMapKey: Key<MutableMap<String, CachedValue<XmlFile>>> = Key.create(
+        "RIDER_SCHEMAS_FILE_MAP_KEY:" + javaClass.name)
+    private var schemas: MutableMap<String, CachedValue<XmlFile>>? = null
```
Do we really need a Map here for only 1 value? `location` will always point to `SCHEMA_LOCATION`

seclerp reviewed (changes requested):
LGTM, thanks! Just a few minor comments.

Commit:
Refactor SpriteFontXmlSchemaProvider for optimization

Simplified handling of cached schema in the `SpriteFontXmlSchemaProvider` class. The extraneous schemas map was removed in favor of having a single cached schema instance. This change reduces complexity while maintaining performance, as the schema is only calculated once and reused as needed.

Alls-DeV committed:
Merge branch 'seclerp:release/232' into fix/17
